python/ray/data/tests/test_streaming_executor.py

```diff
@@ -720,6 +727,54 @@ def patched_get_outupt_blocking(*args, **kwargs):
     assert isinstance(error, KeyboardInterrupt), error
 
 
+def test_execution_callbacks_topology_arg(tmp_path, restore_data_context):
+    """Test the topology arg in ExecutionCallback."""
+
+    topo = None
+
+    class CustomExecutionCallback(ExecutionCallback):
+        def after_execution_succeeds(self, topology: Topology):
+            nonlocal topo
+            topo = topology
+
+    input_path = tmp_path / "input"
+    os.makedirs(input_path)
+    output_path = tmp_path / "output"
+
+    ctx = DataContext.get_current()
+    callback = CustomExecutionCallback()
+    add_execution_callback(callback, ctx)
+    ds = ray.data.read_parquet(input_path)
+
+    def udf(row):
+        return row
+
+    ds = ds.map(udf)
+
+    ds = ds.write_parquet(output_path)
+
+    assert topo is not None
+    assert len(topo) == 2
```

Comment:
Can you add a comment explaining why this should be 2? I don't think it's obvious

Reply:
actually the next 3 lines already explained that